Proposal for an AI policy (#73)
YOURLS/.github: IA_POLICY.md

# AI Usage Policy

The YOURLS project and organization have strict rules regarding AI usage.

- **All AI usage in any form must be disclosed.** You must state

>   the tool you used (e.g. Claude Code) along with the extent that the work

Reviewer: ```suggestion
  the tool you used (e.g. Claude Code) along with the extent to which the work
```

  was AI-assisted.

- **The human-in-the-loop must fully understand all code.** If you can't explain
  without AI what your changes do and how they interact with code, please do not
  contribute to this project.

- **Any content generated with AI must have been reviewed _and edited_

>   by a human before submission**. AI is very good at being overly verbose and

Reviewer: ```suggestion
  by a human before submission.** AI is very good at being overly verbose and
```

Nitpick for consistency with the other bullets.

  including noise that distracts from the main point.

> 

Reviewer: I think it's _Very Important_ that we discourage letting AI write the text of an issue or PR itself. It's kind of implied by the previous points but let's make it crystal clear: We don't want to talk to Claude or Gemini; we want to talk to the person prompting them.

```suggestion

- **The human-in-the-loop must communicate with maintainers.** This follows from
  the preceding points: AI models are verbose and unfocused. Don't have your AI
  tool describe to us what you've found or changed; say it in your own words.

```

- **Pull Requests must be focused.** If you want to submit a PR that fixes a
  problem, enhances or introduce a new feature, make sure your code does just
  that. PR containing also white space changes, rewording comments, or

>   tackling several matters will be rejected.

Reviewer: ```suggestion
  that. PRs that change whitespace, reword comments, or contain multiple
  unrelated sets of changes will be rejected.
```

## There are Humans Here

Please remember that YOURLS is maintained by humans.

Every discussion, issue, and pull request is read and reviewed by
humans. It is a boundary point at which people interact with each other
and the work done. It is rude and disrespectful to approach this boundary
with low-effort, unqualified work, since it puts the burden of validation

> on the maintainer.

Reviewer: ```suggestion
Every discussion, issue, and pull request is read and reviewed by humans.
It is a boundary point at which people interact with each other and guide
the work. It is rude and disrespectful to approach this boundary with
low-effort, unqualified contributions that put the burden of validation
on the maintainers.
```
